ppb_vector/vector2.py

```diff
@@ -59,6 +59,9 @@ class Vector2:
     x: float
     y: float
 
+    # Tell CPython that this isn't an extendable dict
+    __slots__ = ('x', 'y')
```

Comment:
This needs a `'__weakref__'` as well.

Without it, you can't weakly reference a `Vector2`.

Reply:
OK

Reply:
Done